frontend/src/lib/MetricsManager.ts

```diff
@@ -113,9 +121,14 @@ export class MetricsManager {
     return deltaCounter
   }
 
+  public setReportHash = (reportHash: string): void => {
+    this.reportHash = reportHash
+  }
+
   private send(evName: string, evData: object = {}): void {
     const data = {
       ...evData,
+      reportHash: this.reportHash,
```

Comment:
will `reportHash` have a value for every event we send or are there times when we'll send an event but `reportHash` hasn't been set yet? 

Also the opposite situation, is it possible that `reportHash` hasn't been cleared and its old value gets sent?

Reply:
- There is a point when we send an event, but `reportHash` hasn't been set yet. 
- `reportHash` gets cleared in `clearAppState`.

Reply:
> There is a point when we send an event, but reportHash hasn't been set yet.

Cool, it would be good to detail which ones that is

Reply:
> > I believe it should be used as an id in the identify method instead.
> 
> If we use `reportHash` as the `id` in `identify` then the report becomes the user, so I don't think we want that unless I'm missing something.
> 
> Sending as a property of each event, which you're currently doing, seems to be the right move, but I'll let @cobradford confirm that.

You're right. We are loosing the user identity in that case.

Reply:
`viewReport` and `createReport` events don't have enough information to construct a `reportHash`.

Reply:
I think we should capture this information in the MetricsManager docs (possibly in the `reportHash` docstring), because it's very non-obvious from just reading the code:

> `reportHash` gets set when [XYZ] happens. This means that it will be attached to most, but not all, metrics events. The `viewReport` and `createReport` events are sent before [XYZ] happens, so they will not include the `reportHash`.

^ And fill in what "XYZ" is so that a reader of the code can understand when and why reportHash is set (and when and why it's cleared!)